x-pack/test/api_integration/apis/transform/start_transforms.ts

```diff
@@ -0,0 +1,163 @@
+ */
+import expect from '@kbn/expect';
+
+import { StartTransformsRequestSchema } from '../../../../plugins/transform/common/api_schemas/start_transforms';
+
+import { COMMON_REQUEST_HEADERS } from '../../../functional/services/ml/common_api';
+import { USER } from '../../../functional/services/transform/security_common';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+import { asyncForEach, generateDestIndex, generateTransformConfig } from './common';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const transform = getService('transform');
+
+  async function createTransform(transformId: string) {
+    const config = generateTransformConfig(transformId);
+    await transform.api.createTransform(transformId, config);
+  }
+
+  describe('/api/transform/start_transforms', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('ml/farequote');
+      await transform.testResources.setKibanaTimeZoneToUTC();
+    });
+
+    describe('single transform start', function () {
+      const transformId = 'transform-test-start';
+      const destinationIndex = generateDestIndex(transformId);
+
+      beforeEach(async () => {
+        await createTransform(transformId);
+      });
+
+      afterEach(async () => {
+        await transform.api.cleanTransformIndices();
+        await transform.api.deleteIndices(destinationIndex);
+      });
+
+      it('should start the transform by transformId', async () => {
+        const reqBody: StartTransformsRequestSchema = [{ id: transformId }];
+        const { body } = await supertest
+          .post(`/api/transform/start_transforms`)
+          .auth(
+            USER.TRANSFORM_POWERUSER,
+            transform.securityCommon.getPasswordForUser(USER.TRANSFORM_POWERUSER)
+          )
+          .set(COMMON_REQUEST_HEADERS)
+          .send(reqBody)
+          .expect(200);
+
+        expect(body[transformId].success).to.eql(true);
+        expect(typeof body[transformId].error).to.eql('undefined');
+        await transform.api.waitForBatchTransformToComplete(transformId);
+        await transform.api.waitForIndicesToExist(destinationIndex);
+      });
+
+      it('should return 200 with success:false for unauthorized user', async () => {
+        const reqBody: StartTransformsRequestSchema = [{ id: transformId }];
+        const { body } = await supertest
+          .post(`/api/transform/start_transforms`)
+          .auth(
+            USER.TRANSFORM_VIEWER,
+            transform.securityCommon.getPasswordForUser(USER.TRANSFORM_VIEWER)
+          )
+          .set(COMMON_REQUEST_HEADERS)
+          .send(reqBody)
+          .expect(200);
+
+        expect(body[transformId].success).to.eql(false);
+        expect(typeof body[transformId].error).to.eql('string');
+
+        await transform.api.waitForTransformState(transformId, 'stopped');
```

Comment:
Another place where we should use constants out of `TRANSFORM_STATE`

Reply:
Fixed in e434c4f.